rust/src/mime/mod.rs

```diff
@@ -0,0 +1,244 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use std;
+
+use nom::combinator::rest;
+use nom::error::ErrorKind;
+use nom::Err;
+use nom::IResult;
+
+#[derive(Clone, Debug)]
+pub struct MIMEHeaderToken<'a> {
+    pub name: &'a [u8],
+    pub value: &'a [u8],
+}
+
+#[derive(Clone)]
+pub struct MIMEHeaderTokens<'a> {
+    pub tokens: Vec<MIMEHeaderToken<'a>>,
+}
+
+pub fn mime_parse_value_delimited(input: &[u8]) -> IResult<&[u8], &[u8]> {
+    let (i2, _) = tag!(input, "\"")?;
+    let (i3, value) = take_until!(i2, "\"")?;
+    let (i4, _) = tag!(i3, "\"")?;
+    return Ok((i4, value));
```

Comment:
Its OK, and idiomatic to reuse the same variable name if you don't need the previous value, ie)
```
let (input, _) = tag!(input, "\"")?;
let (input, value) = take_until(...
```

Reply:
Ok, but then I need input to be `mut` right ?

Reply:
> Ok, but then I need input to be `mut` right ?

No, shouldn't be.  Redefining is different than mutating.  Rust allowed this for patterns like:
```
let x = Something{}...
// Do some stuff with X..
let x = Arc::new(Mutex::new(x));
// Pass x of to threads or something
```
When dealing with input like above, I think it also fits a pattern where redefining is actually nice.

Reply:
ok done in next version